src/NServiceBus.Core/Hosting/HostCreator.cs

```diff
@@ -51,45 +45,27 @@ public static async Task<IStartableEndpoint> CreateWithInternallyManagedContaine
 
             endpointConfiguration.FinalizeConfiguration(assemblyScanningComponent.AvailableTypes);
 
-            var hostingSetting = settings.Get<HostingComponent.Settings>();
-            var useDefaultBuilder = hostingSetting.CustomObjectBuilder == null;
-            var container = useDefaultBuilder ? new LightInjectObjectBuilder() : hostingSetting.CustomObjectBuilder;
-
-            var commonObjectBuilder = new CommonObjectBuilder(container);
-
-            IConfigureComponents internalContainer = commonObjectBuilder;
-            IBuilder internalBuilder = commonObjectBuilder;
-
-            //for backwards compatibility we need to make the IBuilder available in the container
-            internalContainer.ConfigureComponent(_ => internalBuilder, DependencyLifecycle.SingleInstance);
+            var serviceCollection = new MicrosoftExtensionsDependencyInjection.ServiceCollection();
 
-            var hostingConfiguration = HostingComponent.PrepareConfiguration(settings.Get<HostingComponent.Settings>(), assemblyScanningComponent, internalContainer);
+            var hostingConfiguration = HostingComponent.PrepareConfiguration(settings.Get<HostingComponent.Settings>(), assemblyScanningComponent, serviceCollection);
 
-            if (useDefaultBuilder)
-            {
-                hostingConfiguration.AddStartupDiagnosticsSection("Container", new
-                {
-                    Type = "internal"
-                });
-            }
-            else
+            hostingConfiguration.AddStartupDiagnosticsSection("Container", new
             {
-                var containerType = internalContainer.GetType();
-
-                hostingConfiguration.AddStartupDiagnosticsSection("Container", new
-                {
-                    Type = containerType.FullName,
-                    Version = FileVersionRetriever.GetFileVersion(containerType)
-                });
-            }
+                Type = "internal"
+            });
 
             var endpointCreator = EndpointCreator.Create(settings, hostingConfiguration);
 
             var hostingComponent = HostingComponent.Initialize(hostingConfiguration);
 
-            var startableEndpoint = endpointCreator.CreateStartableEndpoint(internalBuilder, hostingComponent);
+            var containerOptions = new ContainerOptions
+            {
+                EnableVariance = false
```

Comment:
Isn't there also a `EnablePropertyInjection `? Wouldn't we want to set that to `false`?

Reply:
@timbussmann ☝🏼

Reply:
that's already done as part of `WithMicrosoftSettings()` see https://github.com/Particular/NServiceBus/pull/5719/files/6756c65975de2b3e196986824498353d1545db7a?file-filters%5B%5D=.cs&file-filters%5B%5D=.csproj#diff-ce2bef14f55179b47bfb2f914fdfd33bR241